### ISSUE TITLE
[🐛 Bug]: ts-node performs type check regardless of configuration

### ISSUE THREAD

**Issue by GYuriy** [webdriverio/webdriverio#13017]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

latest

### Node.js Version

18.19.1

### Mode

Standalone Mode

### Which capabilities are you using?

```typescript
capabilities: [{
        browserName: 'chrome'
    }],
```

### What happened?

`ts-node` performs type checking regardless of the options (`autoCompile: true,` or `typeCheck: false,`) specified in the wdio config. Alternatively, I tried specifying the `ts-node` options in the tsconfig, but the result is the same, which makes me think it's either a ts-node issue or wdio overrides any options passed to ts-node with default values.

Unfortunately, I can't test on node.js v20+ right now.

### ATTENTION: The attached reproducible example relates to the abovementioned issue.

However, it looks like the issue is older than I might initially think. So, I went back to some of my older codebases and found this setup:
```
    "@types/node": "18.16.14",
    "@wdio/allure-reporter": "8.28.6",
    "@wdio/cli": "8.28.8",
    "@wdio/local-runner": "8.28.8",
    "@wdio/mocha-framework": "8.28.8",
    "@wdio/spec-reporter": "8.28.8",
    "allure-commandline": "2.26.0",,
    "ts-node": "10.9.2",
    "typescript": "5.3.3"
```
In the version above, it works (or actually doesn't work) in the **reverse manner.** The `ts-node` never throws when there are type-related errors in your code. However, unlike the fresh issue aboe, you can achieve a TS error by specifying the ts-node options in the tsconfig.

The will look like this:
```
[0-0] Error: undefined
[0-0]     at MochaAdapter._loadFiles (file:///C:/********/node_modules/@wdio/mocha-framework/build/index.js:91:35)
```
I assume someone was fixing this `Error: undefined` and broke the last ts-node options part that was "working"

### What is your expected behavior?

Tests should run regardless of any type-related errors in the code when `autoCompile: true,` or `typeCheck: false,`

### How to reproduce the bug.

- Init a new TS boilerplate project using `npm init wdio@latest .`
  - Make sure type checking is turned off in ts-node 
- Add the code that will trigger type errors e.g. `let foo: string; foo = 42`
- Run tests with `npm run wdio`
[bug-tsnode-opts.zip](https://github.com/user-attachments/files/15803561/bug-tsnode-opts.zip)

### Relevant log output

```typescript
[0-0] Error: TSError: ⨯ Unable to compile TypeScript:
[0-0] test/specs/test.e2e.ts(5,5): error TS6133: 'foo' is declared but its value is never read.
[0-0] test/specs/test.e2e.ts(9,9): error TS2322: Type 'number' is not assignable to type 'string'.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by erwinheitzman** [comment]
Please check our typescript documentation, these options are not the way to go: https://webdriver.io/docs/typescript#configuration

**Comment by erwinheitzman** [comment]
We have ran without type checking for a while just fine. Use the ts-node options in your tsconfig file or use environment variables

**Comment by GYuriy** [comment]
@erwinheitzman but the issue is that it **always** performs type-checking and I don't want that (you can find the reproducible example attached)
`test/specs/test.e2e.ts(9,9): error TS2322: Type 'number' is not assignable to type 'string'.`

**Comment by erwinheitzman** [comment]
@GYuriy apologies, I misunderstand the issue it seems. So it worked previously but updating webdriverio has broken it?

**Comment by GYuriy** [comment]
@erwinheitzman np. To be honest I've been always using the `transpileOnly` (in my `wdio.config`) set to `true` in all my projects since I always wanted the type-checking. And it was working fine for me since type-check was performed (duh).
But recently I jumped into the project where I need it to be off (not to dig into the reasons). And it appeared that the option in `wdio.config` does not have any effect (as you mentioned). Then I tried to set it in the `tsconfig` - no luck.
Then my QA side curiosity took over and I went to some of my older project and it looks like there were some issues with type-checking on and off for a longer time.

TL;DR I always needed type checking in my project so I never had it turned off. Once I tried to do it I was not able to do it in any of the mentioned ways (wdio config, tsconfig, env var)

**Comment by erwinheitzman** [comment]
I'll try to reproduce this using the latest version when I have some time then 👍

**Comment by GYuriy** [comment]
@erwinheitzman it's reproducible on 8.39.1. Or you meant the alpha?

**Comment by christian-bromann** [comment]
Let's re-open then. As @erwinheitzman suggested, you may want to try the current v9 version which uses `tsx` under the hood and may respect the tsconfig settings.

**Comment by GYuriy** [comment]
@christian-bromann Honestly, I am glad to hear you switched to tsx. By looking at the ts-node repo it feels pretty dead by now...
Thanks for maintaining the project and keeping everything up to date 🙏
